backend/api/chatLink/utils/link.js

```diff
@@ -9,7 +10,9 @@ const generateLink = () => {
     link: `/chat/${hash}`,
     absoluteLink: `${chatLinkDomain}/chat/${hash}`,
     expired: false,
-    deleted: false
+    deleted: false,
+    pin : generatePin(hash, pinLength = 4),
```

Comment:
Although the chances of duplicate are extremely low I would suggest having a check against db before creating new record at https://github.com/muke1908/chat-e2ee/blob/1e92f018aad2b19e4257952d064ce44dd36f1e00/backend/api/chatLink/index.js#L29

Reply:
Sure, I will add this in the next commit

Reply:
Added this functionality:

```javascript
    let link = generateLink();
    //This ensures, PINs won't clash each other
    //Base case loop is not even executed
    //worst case, loop can take 2 or more iterations
    while(findOneFromDB({pin : link.pin}, LINK_COLLECTION)) {
      link = generateLink()
    }
    await insertInDb(link, LINK_COLLECTION);
    return res.send(link);
```